Iteration 2

22DPVEmer opened the pull request:
- Added Savanna.Tests project to solution
- Implemented health and reproduction management for animals
- Added position validation and clamping in GameField
- Updated ConsoleFieldRenderer to handle out-of-bounds positions
- Introduced new constants for health, reproduction, and animal behavior
- Refactored Animal base class to use composition for health and reproduction
- Modified Lion and Antelope classes to support new game mechanics
- No UI changes for now
1. Image of game tests passing:
![image](https://github.com/user-attachments/assets/e7090153-23ae-45f9-b16b-19744239f107)

22DPVEmer committed:
Enhance game mechanics and add test project

- Added Savanna.Tests project to solution
- Implemented health and reproduction management for animals
- Added position validation and clamping in GameField
- Updated ConsoleFieldRenderer to handle out-of-bounds positions
- Introduced new constants for health, reproduction, and animal behavior
- Refactored Animal base class to use composition for health and reproduction
- Modified Lion and Antelope classes to support new game mechanics

sreeharihm commented on Savanna.GameEngine/Models/Antelope.cs:
```diff
@@ -39,13 +43,10 @@ public override void Move(GameField field)
+        protected override void PerformSpecialAction(GameField field)
         {
             // Antelopes don't have special actions yet
+            // Could be extended for features like grazing
```
if they dont have special action, then throw an error to avoid unwanted access

22DPVEmer replied:
Added it and made it also skip the special action in Animal class for antelopes to avoid the code from stopping to run.

sreeharihm commented on Savanna.GameEngine/Models/AntelopeReproductionManager.cs:
```diff
@@ -0,0 +1,71 @@
+            var config = (_parent as Antelope)?.GetConfiguration();
+            if (config == null)
+            {
+                throw new InvalidOperationException("Parent configuration not available");
```
Move exception messages to constants

22DPVEmer replied:
Added it.

22DPVEmer committed:
Refactor test messages and error handling in game engine

- Added TestMessages and ErrorMessages constants to GameConstants
- Updated test cases to use new formatted error and test messages
- Disabled special action for Antelopes in base Animal class
- Implemented NotImplementedException for Antelope's PerformSpecialAction method
- Improved test readability and error reporting

22DPVEmer committed:
Standardize error messages in reproduction managers

- Updated AntelopeReproductionManager and LionReproductionManager to use centralized error message constant
- Replaced hardcoded error message with GameConstants.ErrorMessages.ParentConfigurationNotAvailable

sreeharihm reviewed (approved):
looks good
